### PR TITLE
Added support for angular and linear velocity

### LINKED ISSUE #619: Add Linear/Angular Acceleration/Velocity for XRInputSource
The Gamepad API (used by WebVR) exposes the concept of Linear and Angular Acceleration and Velocities which are not currently included in the Gamepad API in the WebXR spec.  Issue 185 removed these velocity and accelerations for the headsets, but no suitable workarounds exist to get these values from the gamepad.

Including these values would be critical for any developers that want to accurately implement any form of throwing or other gamepad contact even without button presses.

### PR DESCRIPTION
**Pull request by cabanier**
Closes #619

Also silences a couple of bikeshed warnings.

***
<a href="https://pr-preview.s3.amazonaws.com/cabanier/webxr/pull/1182.html" title="Last updated on Mar 13, 2021, 4:42 AM UTC (fb091a2)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/immersive-web/webxr/1182/39aad3f...cabanier:fb091a2.html" title="Last updated on Mar 13, 2021, 4:42 AM UTC (fb091a2)">Diff</a>

### CODE REVIEW

**Comment by cabanier** [comment]
@Manishearth why can I not add reviewers?

File: index.bs
Comment (toji) [inline index.bs:1709]: Angular velocity isn't measured in meters per second. It seems like OpenXR reports this in radians per second, which seems fairly sensible.

**Review by toji — CHANGES_REQUESTED** [review]
LGTM after one correction.

**Comment by toji** [comment]
GitHub had a hangover this morning, which is why you couldn't add reviewers. I think it's generally better now.

**Comment by cabanier** [comment]
> GitHub had a hangover this morning, which is why you couldn't add reviewers. I think it's generally better now.

I still can't add reviewers.

**Comment by cabanier** [comment]
I was looking into a possible implementation and there are some spaces where velocity will never be reported.
For instance XRJointSpaces and XRTargetRaySpace with a XRTargetRayMode different than POINTING.

Should we add those exceptions to the spec, or leave it up to the UA?

**Comment by toji** [comment]
I think it's best to leave that up to the UA. For example, I can see that OpenXR has an extension for [hand joint velocity](https://www.khronos.org/registry/OpenXR/specs/1.0/man/html/openxr.html#_xrhandjointvelocityext3) and I suspect that the tracking method (ie: optical vs sensors) will make a big difference in what velocities are available. I wouldn't want to prematurely prevent future advances in tracking.

If we're concerned about developers needing to know what spaces can and can't report velocities, it would probably be best to report that as a property of the space (`hasLinearVelocity`/`hasAngularVelocity` booleans, maybe?) but I don't know that we can do that consistently and reliably on all platforms, and I believe even spaces that CAN report velocities may have moments where it lapses, such after the first frame or two of tracking (depending on the tracking method). As a result, developers will need to be able to handle `null` velocities no matter what.

We should probably just leave it as "If the user agent can't populate this, it's allowed to return `null`."

**Review by toji — APPROVED** [review]
Thanks for fixing the angular velocity definition, by the way!

**Comment by cabanier** [comment]
@toji thanks for approving! Can you merge it for me? I'm not able to.

**Comment by toji** [comment]
If you don't mind I'd like to make sure that @Manishearth has had a chance to look before I do? That should be Tues at the latest, since we have our editors sync then. Normally I wouldn't be too concerned but given this is a change to a largely "shipped" spec I feel a smidge of extra caution is warranted.

I doubt there's going to be any concerns, though.

**Comment by cabanier** [comment]
I see. That sounds good! There's no hurry to get it in.

**Review by Manishearth — APPROVED** [review]
Seems good now.

One note: angular velocity can be obtained from the other attributes; and we tend towards not having APIs that expose information that can be calculated by the author, but I think it's fine in this case.

**Pull request merged**
